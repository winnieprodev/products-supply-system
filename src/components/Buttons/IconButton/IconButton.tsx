import React from "react";

import * as Styled from "./iconButton.styled";

interface OwnProps {
  primary?: boolean;
  disabled?: boolean;
  transparent?: boolean;
  onClick: () => void;
}

const AmountInput: React.FC<React.PropsWithChildren<OwnProps>> = ({
  primary = false,
  disabled = false,
  transparent = false,
  onClick,
  children,
}) => (
  <Styled.IconButton
    $primary={primary}
    disabled={disabled}
    $transparent={transparent}
    onClick={onClick}
  >
    {children}
  </Styled.IconButton>
);

export default React.memo(AmountInput);
