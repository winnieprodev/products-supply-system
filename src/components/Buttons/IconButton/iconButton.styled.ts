import styled from "styled-components";

import { ButtonCore } from "../buttons.styled";

export const IconButton = styled(ButtonCore)`
  display: inline-flex;
  width: 22px;
  height: 22px;
  font-size: 10px;
  border-radius: 50%;
  align-items: center;
  justify-content: center;
`;
