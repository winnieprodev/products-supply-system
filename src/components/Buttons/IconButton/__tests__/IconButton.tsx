import { render } from "@testing-library/react";

import IconButton from "../IconButton";

describe("IconButton", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <IconButton onClick={jest.fn()}>test</IconButton>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for primary", () => {
      const { container } = render(
        <IconButton primary onClick={jest.fn()}>
          test
        </IconButton>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for disabled", () => {
      const { container } = render(
        <IconButton disabled onClick={jest.fn()}>
          test
        </IconButton>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
