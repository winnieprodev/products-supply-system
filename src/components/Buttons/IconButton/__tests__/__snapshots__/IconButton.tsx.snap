// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Snapshot tests it matches the snapshot 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #EEEEEE;
  color: #5F5F5F;
}

.c0:hover {
  background-color: #E7E7E7;
}

.c0:active {
  background-color: #F3F3F3;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 22px;
  height: 22px;
  font-size: 10px;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<button
  class="c0 c1"
>
  test
</button>
`;

exports[`IconButton Snapshot tests it matches the snapshot for disabled 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #EEEEEE;
  color: #5F5F5F;
  opacity: 0.3;
}

.c0:hover {
  background-color: #E7E7E7;
}

.c0:active {
  background-color: #F3F3F3;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 22px;
  height: 22px;
  font-size: 10px;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<button
  class="c0 c1"
  disabled=""
>
  test
</button>
`;

exports[`IconButton Snapshot tests it matches the snapshot for primary 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
}

.c0:hover {
  background-color: #DD5202;
}

.c0:active {
  background-color: #FF7626;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 22px;
  height: 22px;
  font-size: 10px;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<button
  class="c0 c1"
>
  test
</button>
`;
